crypto-pgpainless/src/main/kotlin/dev/msfjarvis/aps/crypto/KeyUtils.kt

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2014-2022 The Android Password Store Authors. All Rights Reserved.
+ * SPDX-License-Identifier: GPL-3.0-only
+ */
+
+package dev.msfjarvis.aps.crypto
+
+import com.github.michaelbull.result.get
+import com.github.michaelbull.result.runCatching
+import java.util.Locale
+import org.bouncycastle.openpgp.PGPKeyRing
+import org.pgpainless.PGPainless
+
+/** Utility methods to deal with PGP [Key]s. */
+public object KeyUtils {
+  /**
+   * Attempts to parse a [PGPKeyRing] from a given [key]. The key is first tried as a secret key and
+   * then as a public one before the method gives up and returns null.
+   */
+  public fun tryParseKeyring(key: Key): PGPKeyRing? {
```

Comment:
Should we return a `Result<PGPKeyRing>` here?

Reply:
I thought about it but I don't feel there's any value to throwing an exception here.